addons/event_sale/wizard/event_edit_registration.py

```diff
@@ -55,12 +70,18 @@ def default_get(self, fields):
             ('sale_order_id', '=', sale_order.id),
             ('event_ticket_id', 'in', sale_order.mapped('order_line.event_ticket_id').ids),
             ('state', '!=', 'cancel')])
+        so_line_ticket = sale_order.order_line.grouped('event_ticket_id')
+        # Remove so line not related to events
+        if self.env['event.event.ticket'] in so_line_ticket:
+            so_line_ticket.pop(self.env['event.event.ticket'])
 
         attendee_list = []
-        for so_line in [l for l in sale_order.order_line if l.event_ticket_id]:
-            existing_registrations = [r for r in registrations if r.event_ticket_id == so_line.event_ticket_id]
-            for reg in existing_registrations:
-                attendee_list.append([0, 0, {
+        existing_registrations = registrations.grouped(lambda r: (r.event_ticket_id, r.sale_order_line_id))
+        for event_ticket, so_lines in so_line_ticket.items():
+            for so_line in so_lines:
```

Comment:
I feel like I'm missing something but I'd feel dumb not to mention it. Is there something preventing this to work?

- group registrations by so_line
- iterate over so_lines

```
so_lines = sale_order.order_line
so_line_to_reg = registrations.grouped(lambda r: r.sale_order_line_id)
for so_line in so_lines:
               attendee_list += [[0, 0, {
                    'event_id': reg.event_id.id,
                    'event_ticket_id': reg.event_ticket_id.id,
                    'registration_id': reg.id,
                    'phone': reg.phone,
                    'mobile': reg.mobile,
                    'sale_order_line_id': so_line.id,
                }] for reg in so_line_to_reg[so_line]]
                # Add new registrations
                attendee_list += [[0, 0, {
                    'event_id': so_line.event_id.id,
                    'event_ticket_id': so_line.event_ticket_id.id,
                    'sale_order_line_id': so_line.id,
                    'name': so_line.order_partner_id.name,
                    'email': so_line.order_partner_id.email,
                    'phone': so_line.order_partner_id.phone,
                    'mobile': so_line.order_partner_id.mobile,
                }] for count in range(int(so_line.product_uom_qty) - len(so_line_to_reg[so_line]))]
```

Reply:
It's indeed much better, thanks for insisting